ddln_imaginator.proto

```diff
@@ -21,6 +21,64 @@ enum ImageFormat {
     GRAY = 4;
 }
 
+message Environment {
+    message Clouds {
+        enum Type {
+            NONE = 0;
+            ALTOCUMULUS = 1;
+            CLOUDS_BASE = 2;
+            CUMULUS = 3;
+            NIMBOSTRATUS = 4;
+            STRATOCUMULUS = 5;
+            STRATUS = 6;
+        }
+        optional Type type = 1;
+        optional float altitude_m = 2;
+        optional Point3d velocity = 3; // Only use the directional x,y,z values to set the wind in the Engine.
+    }
+    optional Clouds clouds = 1;
+
+    message Haze {
+        enum Type {
+            DEFAULT = 0; // Visibility = 20000, density = 1
```

Comment:
use defaults for visibility and density instead of separate type: https://developers.google.com/protocol-buffers/docs/proto#optional
I don't think you need `Type` enum here

Reply:
This was discussed offline and we decide not to use default values intentionally since they are no longer supported in proto3 (for good reason). Medium term plan is to move towards proto3.

Reply:
LGTM